celery/models/celery_queue.py

```diff
@@ -0,0 +1,75 @@
+# Copyright Nova Code (http://www.novacode.nl)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+from odoo import api, fields, models, _
+from .celery_task import STATE_PENDING, STATE_RETRY, STATE_FAILURE, STATE_SUCCESS
+
+
+class CeleryQueue(models.Model):
+    _name = 'celery.queue'
+    _description = 'Celery Queue'
+    _inherit = ['mail.thread']
+    _order = 'name'
+
+    @api.multi
+    def _get_task_settings(self):
+        for record in self:
+            task_setting_ids = self.env['celery.task.setting.queue'].search([('queue_id', '=', record.id)]) or []
+            if task_setting_ids:
+                task_setting_ids = [t.task_setting_id.id for t in task_setting_ids]
+            record.task_setting_ids = task_setting_ids
+
+    @api.multi
+    def _compute_stats(self):
```

Comment:
@antonioburic those queries (for computed fields) also seem executed from within the `call_task` method (celery task), which returns the ORM object of the `celery.task.setting`.

Could we optimize this with 1 SQL-query which does a UNION?
- Each UNION'd query/SELECT gets an extra column, which implies the specific **stat_field** (pending, 24h, 24_done) - see below, you'll get the idea ;)
- Use 1 `fetchall` which loops through this UNION'd recordset and check which model-object field to set by the **stat_field**.

```
SELECT 
  queue, 
  COUNT(*) AS counted,
  'pending' AS stat_field
FROM celery_task
WHERE state = %s
GROUP BY queue", (STATE_PENDING, )
  UNION
SELECT 
  queue, 
  COUNT(*) AS counted,
  '24h' AS stat_field
FROM celery_task
WHERE create_date > (current_timestamp - interval '24 hour') 
GROUP BY queue
  UNION
...
```

Reply:
@bobslee nice, thanks! I will improve this part. 
I suspected the non-stored computed fields might be triggered at each fetch, but didn't profile it enough to know this is the case. So of course this affects the performance when queues are configured, it will need to be recomputed only when accessing the tree/kanban view.